dispatcher/config.py

```diff
@@ -0,0 +1,77 @@
+import os
+from contextlib import contextmanager
+from typing import Optional
+
+import yaml
+
+from dispatcher.publish import DispatcherDecorator
+from dispatcher.registry import DispatcherMethodRegistry, registry
+from dispatcher.utils import resolve_callable
+
+
+class DispatcherSettings:
+    def __init__(self, config: dict, task_registry: DispatcherMethodRegistry = registry) -> None:
+        self.brokers: list = config.get('brokers', [])
+        self.producers: list = config.get('producers', [])
+        self.service: dict = config.get('service', {'max_workers': 3})
+        self.callbacks: dict = config.get('callbacks', {})
+        self.tasks: dict = config.get('tasks', {})
+        self.settings: dict = config.get('settings', {})
+
+        for task_name, options in self.tasks.items():
+            method = resolve_callable(task_name)
+            if method:
+                DispatcherDecorator(task_registry, **options)(method)
+            else:
+                raise RuntimeError(f'Could not locate task given in config: {task_name}')
+
+
+def settings_from_file(path: str) -> DispatcherSettings:
+    with open(path, 'r') as f:
+        config_content = f.read()
+
+    config = yaml.safe_load(config_content)
+    return DispatcherSettings(config)
+
+
+def settings_from_env() -> DispatcherSettings:
+    if file_path := os.getenv('DISPATCHER_CONFIG_FILE'):
+        return settings_from_file(file_path)
+    raise RuntimeError('Dispatcher not configured, set DISPATCHER_CONFIG_FILE or call dispatcher.config.setup')
+
+
+empty = object()
+
+
+class LazySettings:
+    def __init__(self) -> None:
+        self._wrapped: Optional[DispatcherSettings] = None
+
+    def __getattr__(self, name):
+        if self._wrapped is empty:
```

Comment:
Why not just check if `_wrapped` is none?

Reply:
oh this is a total goof. I abandoned `empty` and this shouldn't be here. Sorry about that. Little premature of a patch.